base/cvd/cuttlefish/host/libs/web/cas/cas_flags.h

```diff
@@ -24,6 +24,40 @@
 
 namespace cuttlefish {
 
+// Wrapper for flag values that tracks whether a value was user-specified
+// (via command-line or config file) versus using a default.
+template <typename T>
+struct FlagValue {
+  T value;
+  bool user_specified = false;
+
+  FlagValue() = default;
+  explicit FlagValue(const T& default_val)
+      : value(default_val), user_specified(false) {}
+
+  // Assignment operator for setting value (preserves user_specified flag)
+  FlagValue& operator=(const T& new_value) {
+    value = new_value;
+    return *this;
+  }
+
+  // Conversion operators for backward compatibility with code expecting T
+  operator const T&() const { return value; }
+  operator T&() { return value; }
+
+  // Pointer operators
+  const T* operator->() const { return &value; }
+  T* operator->() { return &value; }
+
+  // Comparison operators
+  bool operator==(const T& other) const { return value == other; }
+  bool operator!=(const T& other) const { return value != other; }
+  bool operator<(const T& other) const { return value < other; }
+  bool operator>(const T& other) const { return value > other; }
+  bool operator<=(const T& other) const { return value <= other; }
+  bool operator>=(const T& other) const { return value >= other; }
```

Comment:
Are these comparison operators really necessary? It looks like the only container these object are ever added to is a vector.